images/macos/software-report/SoftwareReport.Common.psm1

```diff
@@ -137,424 +152,434 @@ function Get-NVMNodeVersionList {
     $nodejsVersionsRaw = Run-Command "${nvmInitCommand} && nvm ls"
     $nodeVersions = $nodejsVersionsRaw | ForEach-Object { $_.TrimStart(" ").TrimEnd(" *") } | Where-Object { $_.StartsWith("v") }
     $result = [string]::Join(" ", $nodeVersions)
-    return "NVM - Cached node versions: $result"
+    return "$result"
 }
 
 function Build-OSInfoSection {
+    param (
+        [string] $ImageName
+    )
+
     $fieldsToInclude = @("System Version:", "Kernel Version:")
     $rawSystemInfo = Invoke-Expression "system_profiler SPSoftwareDataType"
     $parsedSystemInfo = $rawSystemInfo | Where-Object { -not ($_ | Select-String -NotMatch $fieldsToInclude) } | ForEach-Object { $_.Trim() }
-    $output = ""
     $parsedSystemInfo[0] -match "System Version: macOS (?<version>\d+\.\d+)" | Out-Null
     $version = $Matches.Version
-    $output += New-MDHeader "macOS $version info" -Level 1
-    $output += New-MDList -Style Unordered -Lines $parsedSystemInfo -NoNewLine
-    return $output
+    $systemVersion = $parsedSystemInfo[0].Replace($fieldsToInclude[0],"").Trim()
+    $kernelVersion = $parsedSystemInfo[1].Replace($fieldsToInclude[1],"").Trim()
+
+    $osInfoNode = [HeaderNode]::new("macOS $version info")
+    $osInfoNode.AddToolNode("System Version:", $systemVersion)
+    $osInfoNode.AddToolNode("Kernel Version:", $kernelVersion)
+    $osInfoNode.AddToolNode(("Image Version: {0}" -f $ImageName.Split('_')[1]), "ImageVersion")
```

Comment:
This line seems to produce incorrect output:
`- Image Version: 20221121 ImageVersion`

Should it be?
```pwsh
$osInfoNode.AddToolNode(("Image Version:", $ImageName.Split('_')[1]))
```

Also, may be we can take image version from environment variable? `$env:ImageVersion`?

Reply:
corrected according to the initial approach (did not try option with the environment variable)